WASM.md

```diff
@@ -1,23 +1,68 @@
-Proof of Concept WebAssembly Support.
+WebAssembly Extension Support
 
-# WebAssembly VM
+# Build Dependencies
 
-The WebAssembly VM is https://github.com/WAVM/WAVM. This VM was chosen for the
-proof of concept because:
+## Install glib2.0 if required (on Debian/Ubuntu)
 
-  - It is OSS with a permissive license.
-  - It compiles with LLVM for high performance.
-  - It can attach pre-compiled code to the WASM files (e.g. during distribution
-      from a central controller) for faster startup and reduced load on a pool
-      of proxies.
+apt-get install libglib2.0-dev
 
-# TODO
+## emscripten
 
-  - Check if the thread is running in the VM in the WAVM Runtime signal handlers and
-    if not call the underlying signal handler from Envoy or add the WAVM signal handler
-    as a pre-check to the Envoy signal handler with the same check.
-  - Add support to WAVM for multple stacks for coroutines/streams with thread shared state.
-    - evaluate async emscripten: https://github.com/kripken/emscripten/wiki/Asyncify
-    - evaulate boost::fiber, GNU Pth, getcontext(), setcontext(), swapcontext().
-  - Add support for shared global state.
-  - Intercept logging messages from WAVM and send to Envoy logs.
+git clone https://github.com/emscripten-core/emsdk.git  
+cd emsdk  
+./emsdk install sdk-1.38.25-64bit  
+./emsdk activate sdk-1.38.25-64bit  
+
+. ./emsdk\_env.sh  
+
+It is possible later versions will work, e.g.
+
+./emsdk install latest  
+./emsdk activate latest  
+
+However 1.38.25 is known to work.
+
+## clang-7 or clang-8
+
+export CC=clang  
+export CXX=clang++  
+
+Note: ensure that you have clang in your path (e.g. /usr/lib/llvm-7/bin).
+
+## protobuf v3.6.1
```

Comment:
`protobuf` is a dependency for regenerating test modules, isn't it?

Reply:
It is also a dependency for building your own WebAssembly module which uses any protos (e.g. gRPC which is part of the standard API), so it should be included in the main section.